src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/LegacyPropagator.cs

```diff
@@ -0,0 +1,190 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Diagnostics
+{
+    internal class LegacyPropagator : TextMapPropagator
+    {
+        internal static TextMapPropagator Instance { get; } = new LegacyPropagator();
+
+        public override IReadOnlyCollection<string> Fields { get; } = new HashSet<string>() { TraceParent, RequestId, TraceState, Baggage, CorrelationContext };
```

Comment:
```suggestion
        public override IReadOnlyCollection<string> Fields { get; } = new[] { TraceParent, RequestId, TraceState, Baggage, CorrelationContext };
```

Since we can't use `IReadOnlySet`, a simple array can be used here.

Reply:
I used the hash set just to ensure we'll not add any duplication by mistake. I don't mind changing it to array instead.

Reply:
If we actually want this to be immutable, we should probably use a `ReadOnlyCollection<T>` or something else that can't be mutated when we hand out the reference.  An `ROC<T>` around an array would make sense.